code/lib/types/src/modules/addons.ts

```diff
@@ -301,15 +303,19 @@ export interface Addon_RenderOptions {
   key?: string;
 }
 
+/**
+ * @deprecated This type is deprecated and will be removed in 8.0.
+ */
 export type ReactJSXElement = {
   type: any;
   props: any;
   key: any;
 };
 
-export interface Addon_Type {
-  title: (() => string) | string | ReactJSXElement;
-  type?: Addon_Types;
+export type Addon_Type = Addon_BaseType;
+export interface Addon_BaseType {
```

Comment:
I really don't like the `Addon_Type` and `Addon_BaseType` names here for multiple reasons, but I guess that ship is already sailed.

Reply:
It's all internal API IMHO, so in my mind, the exact name doesn't matter a whole lot.

And I don't think that ship has sailed at all. We can change it if we want.
But I do have a desire to keep the pace high, and changing the names of (internal) interfaces, won't be a blocker for me from moving forwards.

Yes, I know technically users might be importing these types, and we haven't told them not to do that, but I kinda treat the `@storybook/types` package as *mostly* internal, especially the exports that are prefixed such as `Addon_BaseType`.

Users will get these types via the api (`addon.add()` for example).

Reply:
I see. My main gripes are:

1. Mixing PascalCase and snake_case in the same name confuses me, and I don't think it adds value. This is pretty stylistic though.

2. `Addon_Type` (and `Addon_BaseType`) is very close to `Addon_Types`, but they are not the same at all. `Addon_Types` refers to the `type` property, however `Addon_Type` is the whole addon definition. I think it would be better if it was just `Addon` and `BaseAddon`, we usually don't suffix TS types with `_Type` just because they are types. Then we'd also call it `Addon_Loaders_Type`, or heck even `Addon_Types_Type`. 🤪

Feel free to ignore this and move on.